src/mrpro/algorithms/reconstruction/IterativeSENSEReconstruction.py

```diff
@@ -0,0 +1,130 @@
+"""Iterative SENSE Reconstruction by adjoint Fourier transform."""
+
+from __future__ import annotations
+
+from typing import Self
+
+from mrpro.algorithms.optimizers.cg import cg
+from mrpro.algorithms.prewhiten_kspace import prewhiten_kspace
+from mrpro.algorithms.reconstruction.DirectReconstruction import DirectReconstruction
+from mrpro.algorithms.reconstruction.Reconstruction import Reconstruction
+from mrpro.data._kdata.KData import KData
+from mrpro.data.CsmData import CsmData
+from mrpro.data.DcfData import DcfData
+from mrpro.data.IData import IData
+from mrpro.data.KNoise import KNoise
+from mrpro.operators.FourierOp import FourierOp
+from mrpro.operators.LinearOperator import LinearOperator
+
+
+class IterativeSENSEReconstruction(Reconstruction):
+    r"""Iterative SENSE reconstruction.
+
+    This algorithm minizes the problem :math:`min_x \frac{1}{2}||W^\frac{1}{2} (Ax - y)||_2^2`
+    by using a conjugate gradient algorithm to solve
+    :math:`H x = b` with :math:`H = A^H W A` and :math:`b = A^H W y` where :math:`A` is the acquisition model
+    (coil sensitivity maps, Fourier operator, k-space sampling), :math:`y` is the acquired k-space data and :math:`W`
+    describes the density compensation [PRU2001]_ .
+
+    Note: In [PRU2001]_ a k-space filter is applied as a final step to null all k-space values outside the k-space
+    coverage. This is not done here.
+
+    .. [PRU2001] Pruessmann K, Weiger M, Boernert P, and Boesiger P (2001), Advances in sensitivity encoding with
+       arbitrary k-space trajectories. MRI 46, 638-651. https://doi.org/10.1002/mrm.1241
+
+    """
+
+    n_max_iter: int
```

Comment:
Good catch, should be `n_iterations`. I left it at `n_iterations` because IMO it is more likely to use iterative SENSE with a fixed number of iterations rather than being able to set some kind of tolerance.